File: src_f/pfem_init.f
!C
!C***
!C*** PFEM_INIT
!C***
!C
!C    INIT. PFEM-FEM process's
!C
      subroutine PFEM_INIT
      use pfem_util
#ifdef OPENACC
      use openacc
#endif
      implicit REAL*8 (A-H,O-Z)
      integer ngpu, ierr

      call MPI_INIT      (ierr)
      call MPI_COMM_SIZE (MPI_COMM_WORLD, PETOT, ierr )
      call MPI_COMM_RANK (MPI_COMM_WORLD, my_rank, ierr )
      
#ifdef OPENACC
      ngpu = acc_get_num_devices(acc_device_nvidia)
      call acc_set_device_num(mod(my_rank, ngpu), acc_device_nvidia)
#endif

      pfemRarray= 0.d0
      pfemIarray= 0

      return
      end

File: src_c/pfem_init.c
/**
 ** PFEM_INIT
**/
#include "pfem_util.h"

#ifdef OPENACC
#include <openacc.h>
#endif
/**
   INIT. PFEM-FEM process's
**/
void PFEM_INIT(int argc, char* argv[])
{
  int i, ngpu;
  MPI_Init(&argc,&argv);
  MPI_Comm_size(MPI_COMM_WORLD,&PETOT);
  MPI_Comm_rank(MPI_COMM_WORLD,&my_rank);
  
#ifdef OPENACC
  ngpu = acc_get_num_devices(acc_device_nvidia);
  acc_set_device_num(my_rank % ngpu, acc_device_nvidia);
#endif
  
  for(i=0;i<100;i++)  pfemRarray[i]=0.0;
  for(i=0;i<100;i++)  pfemIarray[i]=0;
}
